Adjust CUDA memory leak test

ilia-cher opened the pull request:
Stack from [ghstack](https://github.com/ezyang/ghstack):
* **#40504 Adjust CUDA memory leak test**

Summary:
Make CUDA mem liek test not flaky

Test Plan:
python test/test_profiler.py

Differential Revision: [D22215527](https://our.internmc.facebook.com/intern/diff/D22215527)

Commit:
Adjust CUDA memory leak test

Summary:
Make CUDA mem liek test not flaky

Test Plan:
python test/test_profiler.py

[ghstack-poisoned]

malfet commented on test/test_profiler.py:
```diff
@@ -33,13 +33,15 @@ def test_mem_leak(self):
+            if idx == 0:
+                continue
+            is_increasing = is_increasing and (last_rss[idx] > last_rss[idx - 1])
+        self.assertTrue(not is_increasing,
```
```suggestion
        self.assertTrue(is_increasing,
```

ilia-cher replied:
the condition should be that it is not increasing

malfet replied:
Hmm, then there are definitely a leak:
```
 memory usage is increasing, deque([1951924224, 1951936512, 1951952896, 1951977472, 1951985664], maxlen=5)
```

malfet commented on test/test_profiler.py:
```diff
@@ -33,13 +33,15 @@ def test_mem_leak(self):
-        self.assertTrue(max_diff < 100 * 1024)
+        # with CUDA events leaking the increase in memory was ~7 MB between
+        # profiler invocations above
+        is_increasing = True
+        for idx in range(len(last_rss)):
+            if idx == 0:
+                continue
+            is_increasing = is_increasing and (last_rss[idx] > last_rss[idx - 1])
```
```suggestion
        is_increasing = all([last_rss[idx] > last_rss[idx-1] for idx in range(1, len(last_rss))])
```

Commit:
Update on "Adjust CUDA memory leak test"


Summary:
Make CUDA mem liek test not flaky

Test Plan:
python test/test_profiler.py

[ghstack-poisoned]

Commit:
Update on "Adjust CUDA memory leak test"


Summary:
Make CUDA mem liek test not flaky

Test Plan:
python test/test_profiler.py

[ghstack-poisoned]

Commit:
Update on "Adjust CUDA memory leak test"


Summary:
Make CUDA mem liek test not flaky

Test Plan:
python test/test_profiler.py

[ghstack-poisoned]
